src/scripts/main.js

```diff
@@ -1,3 +1,223 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('button');
+
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const field = document.querySelector('tbody');
+const scoreField = document.querySelector('.game-score');
+
+let state = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+let newState;
+let isWin = false;
+let score = 0;
+const fieldMatrix = [...field.children].map(row => [...row.children]);
+
+button.addEventListener('click', buttonHandler);
+document.addEventListener('keydown', move);
+
+function buttonHandler() {
+  if (button.classList.contains('start')) {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.innerText = 'Restart';
+    messageStart.classList.add('hidden');
+  } else {
+    state = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    score = 0;
+    messageLose.classList.toggle('hidden', true);
+    messageWin.classList.toggle('hidden', true);
+  }
+
+  addTile();
+  addTile();
+  render();
+}
+
+function addTile() {
+  const [y, x] = findEmptyCell();
+
+  state[y][x] = randomNumber();
+}
+
+function findEmptyCell() {
+  const emtyCells = [];
+
+  state.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      if (cell === 0) {
+        emtyCells.push([rowIndex, columnIndex]);
+      }
+    });
+  });
+
+  return emtyCells[Math.floor(Math.random() * emtyCells.length)];
+}
+
+function randomNumber() {
+  return Math.random() >= 0.9 ? 4 : 2;
+}
+
+function render() {
+  state.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      const elem = fieldMatrix[rowIndex][columnIndex];
+
+      elem.classList = 'field-cell';
+
+      if (cell === 0) {
+        elem.innerText = '';
+      } else {
+        elem.innerText = cell;
+        elem.classList.add(`field-cell--${cell}`);
+      }
+    });
+  });
+
+  scoreField.innerText = score;
+}
+
+function move(e) {
+  const Directions = {
+    ArrowLeft: left,
```

Comment:
```suggestion
    ArrowLeft: 'left',
```
and use it in the switch as:
```
case Directions.ArrowLeft:
  left();
```
currently it doesn't work